### PR TITLE
initial PoC

### PR DESCRIPTION
**Pull request by azopat**

### CODE REVIEW

File: upwork-devs/patrick/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+*************************************************************
+```
+
+
+
+## Next Step
+- Validate which sharing technique will be used ( nfs, s3, gcp persistent disk, etc)
+- Implement the updates
+- Plug CI/CD
+
```
Comment (dinis-cruz-gw) [inline upwork-devs/patrick/README.md:114]: Great start 👍

Comment (dinis-cruz-gw) [reply upwork-devs/patrick/README.md:114]: Question: what is inside the 35Mb of the cmd file below? 
(I cant add a comment to that file)

File: upwork-devs/patrick/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"log"
+	"os"
+
+	"github.com/google/uuid"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/filetrust/Open-Source/upwork/project-k8-glasswall-rebuild/pkg/scanner"
```
Comment (dinis-cruz-gw) [inline upwork-devs/patrick/cmd/main.go:11]: What is this doing?

That path doesn’t exist

File: upwork-devs/patrick/pkg/scanner/dispatcher.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package scanner
```
Comment (dinis-cruz-gw) [inline upwork-devs/patrick/pkg/scanner/dispatcher.go:1]: .DS_Store files should not be added to source control

File: upwork-devs/patrick/pkg/scanner/pod_execution.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	input_location := base_path_in_pod + "process-file-" + task_id_str
+	input_location = strings.Replace(input_location, "/", "\\/", -1)
+
+	pod := &kcorev1.Pod{
+		ObjectMeta: kmetav1.ObjectMeta{
+			Name:      "rebuild-" + task_id_str + "-" + job.Batch,
+			Namespace: "test",
+		},
+		Spec: kcorev1.PodSpec{
+			Containers:    []kcorev1.Container{{Name: "rebuild", Image: "azopat/gw-rebuild", Command: []string{"/bin/sh", "-c"}, Args: []string{"sed 's/INPUT_LOCATION/" + input_location + "/g' /home/glasswall/Config.ini.tmpl > /home/glasswall/Config.ini; glasswallCLI -config=/home/glasswall/Config.ini -xmlconfig=/home/glasswall/Config.xml && sleep infinity"}}},
```
Comment (dinis-cruz-gw) [inline upwork-devs/patrick/pkg/scanner/pod_execution.go:34]: Refactor this is make it less like a hack

File: upwork-devs/patrick/pkg/scanner/scan_processor.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+type ScanProcessor struct {
+	Batch  string
+	Folder string
+}
+
+func (s *ScanProcessor) ScanFiles() {
+
+	log.Println("Scan processor on folder " + s.Folder)
```
Comment (dinis-cruz-gw) [inline upwork-devs/patrick/pkg/scanner/scan_processor.go:15]: What is a good way to capture and visualise the logs in a central way?

**Review by dinis-cruz-gw — COMMENTED** [review]
Great PR

Can you answer the questions before I merge it?

File: upwork-devs/patrick/README.md (hunk 1 above)
Comment (azopat) [reply upwork-devs/patrick/README.md:114]: oh yes good catch, that cmd shouldn't be included in the PR. it's the binary file of the go code when i test locally. will delete and add to .gitignore

File: upwork-devs/patrick/pkg/scanner/scan_processor.go (hunk 5 above)
Comment (azopat) [reply upwork-devs/patrick/pkg/scanner/scan_processor.go:15]: We can include a sidecard container that will export all these Println to a central log location which you use.

File: upwork-devs/patrick/pkg/scanner/pod_execution.go (hunk 4 above)
Comment (azopat) [reply upwork-devs/patrick/pkg/scanner/pod_execution.go:34]: ok will do

File: upwork-devs/patrick/pkg/scanner/dispatcher.go (hunk 3 above)
Comment (azopat) [reply upwork-devs/patrick/pkg/scanner/dispatcher.go:1]: true, will include to .gitignore

File: upwork-devs/patrick/cmd/main.go (hunk 2 above)
Comment (azopat) [reply upwork-devs/patrick/cmd/main.go:11]: it's importing the libs. it's managed by go modules, if you are talking about - github.com/filetrust/Open-Source/upwork/project-k8-glasswall-rebuild/pkg/scanner , check go.mod file where i defined the base. it used it to get the correct path

**Comment by azopat** [comment]
@dinis-cruz-gw   it's ready again for review

**Pull request merged**